src/matlab2tikz.m

```diff
@@ -2378,6 +2395,9 @@ case size(cData,2) % specific y-data is given
             end
         case 3    % Text within a 3d plot
             pos = applyHgTransform(m2t, pos);
+            
+            %3d axes (need to set this for axes that contain only text
+            m2t.currentAxesContain3dData=true;
```

Comment:
We are not using that approach for 3D axis. We have now a direct check with the method `isAxis3D()`. Basically, these 3 lines are unnecessary.